pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -358,7 +358,13 @@ func receiveThread(con *XdsConnection, reqChannel chan *xdsapi.DiscoveryRequest,
 			totalXDSInternalErrors.Add(1)
 			return
 		}
-		reqChannel <- req
+		select {
+		case <-con.stream.Context().Done():
```

Comment:
Hmm, there's also con.doneChannel, which depending on what happens is closed slightly after or slightly before the context is done. I wonder if we should just get rid of it and always use Context().Done()?

BTW there's also https://github.com/istio/istio/pull/10862

Reply:
of course, we can.